### DISCUSSION TITLE
notes from setting up and using cosmic DE 2 hours

### DISCUSSION THREAD

**Discussion by thor314** [pop-os/cosmic-epoch#1108] [General]
I tried out cosmic for the last two hours and i'm going to stick with it for now. I wanted to share my notes in case they were helpful to developers.

As of 2024-10-10, CDE has some rough edges, particularly around custom keybinds, GNOME-like alt-tab and alt-~ behavior is not yet implemented, and tdrop/guake-like behavior is on the epoch 2 feature list. Not all settings panel options are implemented yet.

Wayland is fast (much faster than Xorg), the DE has nice tiling and animations (I would like them to be a little faster), is at least good enough to daily drive. 

My issues generally appear to have already been noted and are at least on a development cycle to do list.

I would ilke better documentation for diving in, but I'm going to daily drive CDE for now at least.

## first pass issues
- tdrop is designed to run on X11, and does not work on wayland (or by extension, cosmic). This might require n>2hrs development, and seems like a candidate for an open source contributor. [Some issues](https://github.com/pop-os/cosmic-epoch/issues/603) [have](https://github.com/pop-os/cosmic-epoch/issues/121) been made, and it looks like it's on cosmic epoch 2 for development.
- I would like to have my alt-tab and alt-~ behavior back. [issue](https://github.com/pop-os/cosmic-launcher/issues/135). On epoch 1 dev cycle.
- [keybinds reset with a config file](https://gitlab.com/theshatterstone/dotfiles/-/blob/e96536882a208bbc1e335865740395743d4289c1/Cosmic/etc/cosmic-comp/config.ron) - can't just press the keybind in settings modal; have to type it out - my file doesn't seem to actually get loaded, even after a logout login, so I'm not getting my custom keybinds loaded. 
- first log out resulted in blackscreen, ctrl+alt+f2 fixed this. 

## setup log 
- Trying cosmic greeter instead of gdm. Unsure how I will unset this if needed.
- initial observations
    - initially, obsidian is overscaled, but we can fix that in displays.
    - keybindings and caps ctrl are unbound as expected. welcome to wayland and cosmic.
    - I see less, or rather, no lag in the system picker, or in Obsidian as compared to Gnome/Xorg.
    - blue outline on active window is a little annoying, can change its color to be less intrusive. I might turn it off altogether if I can find that, or make it lighter - got it, and smaller gaps
    - they've implemented nice tiling
    - some animations are implemented
    - clipboard may issues? - seemingly not
    - alt tab behaves differently, huh
    - curious if they've got notifications figured out yet
    - need asap: [caps lock to ctrl](https://github.com/pop-os/cosmic-epoch/issues/991) - modified `.config/cosmic/com.system76.CosmicComp/v1/xkb_config`
- settings panel
    - no observable way to turn off dock - just kidding, under desktop.
    - no option to set caps lock to ctrl in settings yet
    - some panels, like users, not implemented yet
    - option to hide top bar, nice
    - move windows looks great
    - hope they implement a way to export all settings, esp keybinds

**Comment by mmstick** [comment]
To export all settings, copy `~/.config/cosmic/`
